app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,83 @@
+class VendorsController < ApplicationController
+  def index
+    uri = URI(rmapi_path)
+
+    # Transform query params to what the EBSCO RM-API expects
+    uri.query = URI.encode_www_form(
+      search: params[:q],
+      orderby: params[:orderby] || params[:q] ? 'relevance' : 'vendorname',
+      count: params[:count] || 25,
+      offset: params[:offset] || 1
+    )
+
+    # Make the RM-API request and parse the response body
+    response = make_request(uri)
+    data = JSON.parse(response.body)
+
+    # Return the list of vendors
+    if response.code === '200'
+      vendors = data['vendors'].map do |vendor|
+        Vendor.new(vendor)
+      end
+
+      render jsonapi: vendors,
+             meta: { totalResults: data['totalResults'] },
+             status: :ok
+    else
+      # Map RM-API errors to JSON-API error objects
+      errors = data['Errors'].map { |err| { title: err['Message'] } }
+      render jsonapi_errors: errors, status: response.code
+    end
+  end
+
+  def show
+    uri = URI(
+      "#{rmapi_path}/%{vendor_id}" % {
+        vendor_id: params[:id]
+      }
+    )
+
+    # Make the RM-API request and parse the response body
+    response = make_request(uri)
+    data = JSON.parse(response.body)
+
+    # Return the requested vendor
+    if response.code === '200'
+      vendor = Vendor.new(data)
+      render jsonapi: vendor, status: :ok
+    else
+      # Map RMAPI errors to JSON-API error objects
+      errors = data['Errors'].map { |err| { title: err['Message'] } }
```

Comment:
small as it is, this is starting to look like something that we'll be repeating enough to warrant its own lil method or something